circuits/circom/processMessages.circom

```diff
@@ -4,500 +4,781 @@ pragma circom 2.0.0;
 include "./mux1.circom";
 
 // local imports
-include "./hasherSha256.circom";
-include "./messageHasher.circom";
-include "./messageToCommand.circom";
-include "./privToPubKey.circom";
-include "./stateLeafAndBallotTransformer.circom";
-include "./trees/incrementalQuinTree.circom";
-include "./utils.circom";
-
-// Proves the correctness of processing a batch of messages.
+include "./utils/hashers.circom";
+include "./utils/messageToCommand.circom";
+include "./utils/privToPubKey.circom";
+include "./utils/stateLeafAndBallotTransformer.circom";
+include "./trees/incrementalQuinaryTree.circom";
+// zk-kit imports
+include "./safe-comparators.circom";
+
+/**
+ * Proves the correctness of processing a batch of MACI messages.
+ * The msgBatchDepth parameter is known as msgSubtreeDepth and indicates the depth 
+ * of the shortest tree that can fit all the messages in a batch.
+ * This template supports the Quadratic Voting (QV).
+ */
 template ProcessMessages(
     stateTreeDepth,
     msgTreeDepth,
-    msgBatchDepth, // aka msgSubtreeDepth
+    msgBatchDepth,
     voteOptionTreeDepth
 ) {
-    // stateTreeDepth: the depth of the state tree
-    // msgTreeDepth: the depth of the message tree
-    // msgBatchDepth: the depth of the shortest tree that can fit all the
-    //                  messages in a batch
-    // voteOptionTreeDepth: depth of the vote option tree
-
-    // we want to ensure that the trees have a valid structure
+    // Must ensure that the trees have a valid structure.
     assert(stateTreeDepth > 0);
     assert(msgBatchDepth > 0);
     assert(voteOptionTreeDepth > 0);
     assert(msgTreeDepth >= msgBatchDepth);
 
-    // default to quinary merkle tree
+    // Default for IQT (quinary trees).
     var TREE_ARITY = 5;
     var batchSize = TREE_ARITY ** msgBatchDepth;
-
     var MSG_LENGTH = 11;
     var PACKED_CMD_LENGTH = 4;
-
     var STATE_LEAF_LENGTH = 4;
     var BALLOT_LENGTH = 2;
-
     var BALLOT_NONCE_IDX = 0;
     var BALLOT_VO_ROOT_IDX = 1;
-
     var STATE_LEAF_PUB_X_IDX = 0;
     var STATE_LEAF_PUB_Y_IDX = 1;
     var STATE_LEAF_VOICE_CREDIT_BALANCE_IDX = 2;
     var STATE_LEAF_TIMESTAMP_IDX = 3;
+    var msgTreeZeroValue = 8370432830353022751713833565135785980866757267633941821328460903436894336785;
 
-    var N_BITS = 252;
```

Comment:
why remove N_BITS here? I think we still need to check the amount?

Reply:
you are right. I missed more of them while rebasing back. I will amend on the next commits.